ank/src/cli_commands/set_state.rs

```diff
@@ -117,9 +168,192 @@ impl CliCommands {
+        let mut complete_state = CompleteState {
+            desired_state: State {
+                workloads: HashMap::from([("nginx".to_string(), workload_spec)]),
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+        let mut complete_state_object: Object = complete_state.try_into().unwrap();
+        let value: serde_yaml::Value = serde_yaml::from_str(SAMPLE_CONFIG).unwrap();
+        let temp_object = Object::try_from(&value).unwrap();
+        let update_mask = vec!["desiredState.workloads.nginx".to_string()];
+
+        overwrite_using_field_mask(&mut complete_state_object, &update_mask, &temp_object);
+
+        complete_state = complete_state_object.try_into().unwrap();
+
+        assert!(complete_state.desired_state.workloads.contains_key("nginx"));
+        assert!(
+            complete_state
+                .desired_state
+                .workloads
+                .get("nginx")
+                .unwrap()
+                .restart_policy
+                == RestartPolicy::Always
+        )
+    }
+
+    // [utest->swdd~cli-supports-yaml-to-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_process_inputs_stdin() {
+        let input = SAMPLE_CONFIG;
+        let reader = Cursor::new(input);
+        let state_object_file = "-".to_string();
+        let mut temp_obj = Object::default();
+
+        process_inputs(reader, &state_object_file, &mut temp_obj).await;
+
+        let value: Value = serde_yaml::from_str(SAMPLE_CONFIG).unwrap();
+        let expected_obj = Object::try_from(&value).unwrap();
+
+        assert_eq!(temp_obj, expected_obj);
+    }
+
+    // [utest->swdd~cli-supports-yaml-to-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_process_inputs_file() {
+        let state_object_file = SAMPLE_CONFIG.to_owned();
+        let mut temp_obj = Object::default();
+
+        process_inputs(io::empty(), &state_object_file, &mut temp_obj).await;
+
+        let value: Value = serde_yaml::from_str(SAMPLE_CONFIG).unwrap();
+        let expected_obj = Object::try_from(&value).unwrap();
+
+        assert_eq!(temp_obj, expected_obj);
+    }
+
+    // [utest->swdd~cli-supports-yaml-to-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_process_inputs_invalid_yaml() {
+        let input = "invalid yaml";
+        let reader = Cursor::new(input);
+        let state_object_file = "-".to_string();
+        let mut temp_obj = Object::default();
+
+        process_inputs(reader, &state_object_file, &mut temp_obj).await;
+    }
+
+    // [utest->swdd~cli-provides-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_set_state_ok() {
+        let update_mask = vec!["desiredState.workloads.nginx.restartPolicy".to_string()];
+        let state_object_file = Some(SAMPLE_CONFIG.to_owned());
+
+        let mut workload_spec = StoredWorkloadSpec::default();
+        workload_spec.restart_policy = RestartPolicy::Always;
```

Comment:
This produces RustAnalyzer warnings, if have a default constructed and a field assigned afterwards.

```suggestion
        let workload_spec = StoredWorkloadSpec {
            restart_policy: RestartPolicy::Always,
            ..Default::default()
        };
```